src/test/java/uk/gov/pay/products/service/PaymentCreatorTest.java

```diff
@@ -86,10 +95,76 @@ public void shouldCreateASuccessfulPayment() throws Exception {
         PaymentResponse paymentResponse = createPaymentResponse(
                 paymentId,
                 paymentAmount,
-                paymentNextUrl);
+                paymentNextUrl,
+                productReturnUrl);
+
+
+        when(productDao.findByExternalId(productExternalId)).thenReturn(Optional.of(productEntity));
+        when(RandomIdGenerator.randomUuid()).thenReturn(paymentExternalId);
```

Comment:
For this test you are not even using this randomUuid() for anything. So why bother power mocking?

Reply:
Because we mock `RandomIdGenerator` in `setup()` we need to tell it what to return when running the test otherwise `PaymentCreator.beforePaymentCreation(...)` fails to assign an external id. I can move it to the mock within the test method I assume.